### ISSUE TITLE
can no longer delete namespaces after upgrading from v1.8.3 to v1.9

### ISSUE THREAD

**Issue by rosskirkpat** [rancher/terraform-provider-rancher2#438]
After upgrading from v1.8.3 (where namespaces are successfully destroyed) to v1.9 of the rancher2 TF provider, destroying namespaces is now timing out.

There were no other changes made to the TF script besides changing the version number of the provider. These actions are performed using the admin credentials created by the rancher2 bootstrap. In the Rancher UI, these namespaces become stuck on the "Removing" status.

screenshot attached of debug output from v1.9.0 timeouts when attempting to destroy namespaces

![image](https://user-images.githubusercontent.com/55704795/91613465-90d73d00-e94d-11ea-814f-9b44f08c1c9e.png)

gz#11485

**Comment by rawmind0** [comment]
The provider is waiting for the namespace to get removed, this is the expected behaviour. If the namespace is stuck on the "removing" status in the UI, k8s should be waiting to something finish within the namespace. Are you deploying something different on this namespace?? Same k8s version??

**Comment by rosskirkpat** [comment]
@rawmind0 The only change made was bumping the provider version from 1.8.3 and 1.9.0. This was for an automated build environment. Rancher version and k8s versions were the same between 1.8.3 and 1.9.0.

**Comment by rawmind0** [comment]
The provider will wait until the namespace is removed from Rancher, this is the expected behaviour. For some reason your namespace is stuck on "removing" status at Rancher. Have you took a look to the namespaces where it's stuck??

@rosskirkpat , your issue should be related with https://github.com/kubernetes/kubernetes/issues/60807

**Comment by sriddell** [comment]
This may not be the same problem, but i ran into something similar upgrading to 1.9.0 or beyond.  I tracked it down to the api for  v1beta1.metrics.k8s.io not being ready, because no metrics-service pods were running.

In my rke_config, I just had

monitoring {
  provider = "metrics-server"
}

Before 1.9.0, the replicas default appeared to be 1.

After 1.9.0, the default replicas appeared to be 0.  I had to change my monitoring block to be

monitoring {
  provider = "metrics-server"
  replicas = 1
}

in order to get the metrics server to run and the v1beta1.metrics.k8s.io to be available, which allowed namespaces to be terminated cleanly again.
